app/models/user_mode/auth0_auth_strategy.rb

```diff
@@ -0,0 +1,22 @@
+class UserMode::Auth0AuthStrategy < UserMode::MultiUser
+  def set_auth_provider(omniauth)
+    omniauth.provider :auth0,
+                      Rails.configuration.auth0_client_id,
+                      Rails.configuration.auth0_client_secret,
```

Comment:
Another option is to rename this whole thing from `user_mode` to `authorization_strategy`, which possible values: 

* `no-auth`
* `auth0`
* `saml`